api/logics.py

```diff
@@ -1009,10 +1009,17 @@ def cancel_order(cls, order, user, state=None):
         if order.status in do_not_cancel:
             return False, {"bad_request": "You cannot cancel this order"}
 
-        # 1) When maker cancels before bond
+        # 1) If the order status is not the one specified by the user, do not cancel the order.
+        if cancel_status is not None and order.status != cancel_status:
+            return False, {
+                "bad_request": f"Order status {order.status} is not {cancel_status}. " +
+                "The order may have been taken before it was cancelled"
```

Comment:
This message `"The order may have been taken before it was cancelled"` only applies to a very specific case of `order.status != cancel_status` . Maybe best remove it. Very often this bad_request will be returned because of a check between order status that is not `public != taken`.

Reply:
Yes agree, also I think it is better to put this check before the `if order.status in do_not_cancel` like in my pull request, because otherwise if the client has a wrong `cancel_status`, it gets returned "You cannot cancel this trade", but this is referring to the correct order status. I may confuse a bit the client, it is better to first check if the `cancel_status` is correct.